lib/logstash/inputs/twitter.rb

```diff
@@ -98,11 +99,24 @@ class LogStash::Inputs::Twitter < LogStash::Inputs::Base
   # Port where the proxy is listening, by default 3128 (squid)
   config :proxy_port, :validate => :number, :default => 3128
 
+  # Username where the proxy is listening, by default 3128 (squid)
+  config :proxy_port, :validate => :number, :default => 3128
+
+  # Proxy user
+  config :proxy_user, :validate => :string
+
+  # Proxy user password
+  config :proxy_password, :validate => :password
+
   # Duration in seconds to wait before retrying a connection when twitter responds with a 429 TooManyRequests
   # In some cases the 'x-rate-limit-reset' header is not set in the response and <error>.rate_limit.reset_in
   # is nil. If this occurs then we use the integer specified here. The default is 5 minutes.
   config :rate_limit_reset_in, :validate => :number, :default => 300
 
+
+  # Duration in seconds to wait before retrying when Twitter client socket raises a EOF on network interruption
+  config :steam_eof_wait_duration, :validate => :number, :default => 30
```

Comment:
should we just implement an exponential backoff with no need for more configuration items?

Reply:
Now, if only we had a exponential backoff widget.
I will add the backoff.